### ISSUE TITLE
The balance is unavailable/Cannot detect funds while offline

### ISSUE THREAD

**Issue by fahimaltinordu** [woodser/moneypacket.org#5]
Hello, i am newbie, 
just forked for ILCoin, everything is okey, however the status is **offline**
what i did wrong, can you help me?
Kind regards

**Comment by woodser** [comment]
The tool gives that status if it cannot reach https://bitpay.com/api/rates to detect funds.

Can your browser reach that URL?

Are there any errors or warnings in your browser console?

**Comment by fahimaltinordu** [comment]
I tried coinmarketcap api

- https://api.coinmarketcap.com/v1/ticker/ilcoin/,

With

function get_exchange_rate(code) {

	if (code == "ILC") return 1.0;

	if (code == "mILC") return 1000.0;

	if (code == "bits") return 1000000.0;

	for (var i = 0; i < exchange_rates.length; i++) {

		if (exchange_rates[i].symbol == code) return exchange_rates[i].price_usd;

	}

	return null;

However this time the the buttons dont work, you can see here: https://plusbit.tech/DropCoin-master/

**Comment by woodser** [comment]
The browser console indicates an error in controller.js line 1298.  Looks like a missing bracket to close the if conditional.

**Comment by fahimaltinordu** [comment]
many thanks, bracket is added and buttons work now.
However it is still offline, i think it is about api, although it works on my browser. (https://api.coinmarketcap.com/v1/ticker/ilcoin/ ; https://api.coingecko.com/api/v3/simple/price?ids=ilcoin&vs_currencies=usd%2C)

- The balance is unavailable. 
- Cannot detect funds while offline
- Once i enter optionally an amount to load: $; the result is NaN ILC

**Comment by woodser** [comment]
This tool would need updated to use coinmarketcap's API which is different from bitpay.com's API to query prices.

Additionally, this tool uses bitcore.js to send BTC transactions and check balances, so that library would need replaced with similar logic for your desired coin.

**Comment by fahimaltinordu** [comment]
Yes i edited bitcore.js according to desired coin, just missing dnsSeed, maybe this cause a problem, i dont know

addNetwork({
	name: 'livenet',
	alias: 'mainnet',
	pubkeyhash: 0x00,
	privatekey: 0x80,
	scripthash: 0x05,
	xpubkey: 0x0488b21e,
	xprivkey: 0x0488ade4,
	networkMagic: 0xf9beb4d9,
	port: 6666,
	dnsSeeds: [
	''
	]
});

Regarding coinmarketcap issue, i should study on it.

Many thanks for your help
Kind regards

**Comment by fahimaltinordu** [comment]
Hello Woodser, sorry for disturb again,
i couldnot fix, the coinmarketcap part look okey for me but still have a problem. 
can you please help me?
https://github.com/PlusBitPos/moneypacket.org
Kind regards

**Comment by fahimaltinordu** [comment]
What happens if i completely remove price part?
